skimage/restoration/inpainting.py

```diff
@@ -0,0 +1,183 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+import skimage
+from scipy import sparse
+from scipy.sparse.linalg import spsolve
+
+
+def biharmonic_inpaint(img, mask):
+    # XXX: check and/or rework examples
+    """Inpaint masked points in image, using system of biharmonic
+    equations.
+
+    Parameters
+    ----------
+    img: 2-D ndarray
+        Input image.
+    mask: 2-D ndarray
+        Array of pixels to be inpainted. Has to have the same size as 'img'.
+        Unknown pixels has to be represented with 1, known - with 0.
+
+    Returns
+    -------
+    out: 2-D ndarray
+        Input image with masked pixels inpainted.
+
+    Example
+    -------
+    # >>> import numpy as np
+    # >>> from skimage.restoration.inpainting import biharmonic_inpaint
+    # >>> image_in = np.ones((5, 5))
+    # >>> image_in[:, :2] = 1
+    # >>> image_in[:, 2]  = 2
+    # >>> image_in[:, 3:] = 3
+    # >>> image_in
+    # array([[ 1.,  1.,  2.,  3.,  3.],
+    #        [ 1.,  1.,  2.,  3.,  3.],
+    #        [ 1.,  1.,  2.,  3.,  3.],
+    #        [ 1.,  1.,  2.,  3.,  3.],
+    #        [ 1.,  1.,  2.,  3.,  3.]])
+    # >>> mask = np.zeros_like(image_in)
+    # >>> mask[1:3, 2:] = 1
+    # >>> mask
+    # array([[ 0.,  0.,  0.,  0.,  0.],
+    #        [ 0.,  0.,  1.,  1.,  1.],
+    #        [ 0.,  0.,  1.,  1.,  1.],
+    #        [ 0.,  0.,  0.,  0.,  0.],
+    #        [ 0.,  0.,  0.,  0.,  0.]])
+    # >>> image_in = image_in + mask * 100
+    # >>> image_in
+    # array([[ 1.,  1.,    2.,    3.,    3.],
+    #        [ 1.,  1.,  102.,  103.,  103.],
+    #        [ 1.,  1.,  102.,  103.,  103.],
+    #        [ 1.,  1.,    2.,    3.,    3.],
+    #        [ 1.,  1.,    2.,    3.,    3.]])
+    # >>> image_out = biharmonic_inpaint(image_in, mask)
+    # >>> image_out
+    # array([[ 1.,  1.,  2.,  3.,  3.],
+    #        [ 1.,  1.,  2.,  3.,  3.],
+    #        [ 1.,  1.,  2.,  3.,  3.],
+    #        [ 1.,  1.,  2.,  3.,  3.],
+
+    References
+    ----------
+    Algorithm is based on:
+    .. [1]  N.S.Hoang, S.B.Damelin, "On surface completion and image inpainting
+            by biharmonic functions: numerical aspects",
+            http://www.ima.umn.edu/~damelin/biharmonic
+
+    Realization is based on:
+    .. [2]  John D'Errico,
+        http://www.mathworks.com/matlabcentral/fileexchange/4551-inpaint-nans,
+            method 3
+    """
+
+    # TODO: add sufficient conditions (e.g. unknown area has to be <= 1/16)
+
+    img = skimage.img_as_float(img)
```

Comment:
Is it easy to just to a img.min() and img.max() and then use those values in the algorithm rather than the fixed 0, 1?